File: pset6/caesar.py
import cs50
import sys

if len(sys.argv)!=2:
    print("Please input integer in command line.")
    
    
else:
    k =int(sys.argv[1])
    while True:
        print("Please input a non-negative integer in command line.")
        if k>=0 and str.isdigit(sys.argv[1]):
            break
        
    print("plaintext: ", end="")
    x= cs50.get_string()
    print("ciphertext: ", end="")
    for i in range(len(x)):
        if (str.isalpha(x[i])):
            if (str.isupper(x[i])):
                if ord(x[i])+k<=ord('Z'):
                    y= ord(x[i])+k
                    #str(chr()) converts int to char with corresponding ASCII value Ref:https://stackoverflow.com/questions/3673428/convert-int-to-ascii-and-back-in-python
                    y=str(chr(y))
                    print(y, end="")
                
                else:
                    yupper = (((ord(x[i]))-65+k)%26)+65
                    yupper=str(chr(yupper))
                    print(yupper, end="")
            
    
    
            elif(str.islower(x[i])):
                if ord(x[i])+k<=ord('z'):
                    ylower= ord(x[i])+k
                    ylower=str(chr(ylower))
                    print(ylower, end="")
                
            
                else: 
                    ylower2 = ((ord(x[i])-97+k)%26)+97
                    ylower2 = str(chr(ylower2))
                    print(ylower2, end="")
            
    
        else: 
            print(x[i], end="")
    
    

print("\n")
    
   
 
 

File: pset2/caesar/caesar.c
#include <stdlib.h>
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
int main(int argc, string argv[])
{
    if (argc!=2) {
        printf("Please input integer in command line.\n");
        return 1;
    }

    else {
        int k =atoi(argv[1]);
        if (k < 0) {
         printf("Please input a non-negative integer in command line.\n");
        return 1;
        }
        printf("plaintext: ");
        string x= get_string();
        printf("ciphertext:");
    for (int i = 0, n = strlen(x); i < n; i++) {

        if (isalpha(x[i])) {
            if (isupper(x[i])) {
                if(x[i]+k<='Z') {
                    char y= x[i]+k;
                    printf("%c", y);
                }
            else {
                //k=(x[i]+k)%26;
                char yupper = ((x[i]-65+k)%26)+65;
                printf("%c", yupper);
            }

    }
        else if(islower(x[i])) {
             if(x[i]+k<='z') {
                char ylower= x[i]+k;
                printf("%c", ylower);

            }
            else {
                char ylower2 = ((x[i]-97+k)%26)+97;
                printf("%c", ylower2);

            }

        }

    }
    else {
        printf("%c",x[i]);
    }
    }
}
  printf("\n");
  return 0;
    }

